astropy/utils/tests/test_data.py

```diff
@@ -1055,16 +1055,22 @@ def osraiser(dirnm, linkto, pkgname=None):
     with pytest.warns(CacheMissingWarning) as warning_lines:
         fnout = download_file(TESTURL, cache=True)
     n_warns = len(warning_lines)
-    if n_warns == 2:
-        idx_w1 = 0
-        idx_w2 = 1
-    elif n_warns == 4:  # Two mysterious unclosed socket warnings sometimes
-        idx_w1 = 0
-        idx_w2 = 3
-    else:
-        raise AssertionError(f'Expected 2 or 4 warnings, got {n_warns}')
-    assert 'Remote data cache could not be accessed' in str(warning_lines[idx_w1].message)
-    assert 'temporary' in str(warning_lines[idx_w2].message)
+
+    assert n_warns in (2, 4), f'Expected 2 or 4 warnings, got {n_warns}'
+
+    warning_msgs = [wl.message for wl in warning_lines]
```

Comment:
Not sure if you need to explicitly cast it to str or not. I usually do.
```suggestion
    warning_msgs = [str(wl.message) for wl in warning_lines]
```